internal/pkg/service/stream/mapping/table/column/renderer_test.go

```diff
@@ -75,6 +76,166 @@ func TestRenderer_Headers(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{
+		Path: "key1.key2",
+	}
+
+	body := `{"key1":{"key2":"val2"},"key3":"val3"}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, "\"val2\"", val)
+}
+
+func TestRenderer_Path_Json_Object(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{
+		Path: "key1",
+	}
+
+	body := `{"key1":{"key2":"val2"},"key3":"val3"}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, `{"key2":"val2"}`, val)
+}
+
+func TestRenderer_Path_Json_ArrayOfObjects(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{
+		Path: "key1",
+	}
+
+	body := `{"key1":[{"key2":"val2","key3":"val3"}]}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, `[{"key2":"val2","key3":"val3"}]`, val)
+}
+
+func TestRenderer_Path_Json_ArrayIndex(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{
+		Path: "key1[1].key3",
+	}
+
+	body := `{"key1":[{"key2":"val2"},{"key3":"val3"}]}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, `"val3"`, val)
+}
+
+func TestRenderer_Path_Json_ArrayIndex_RawString(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{
+		Path:      "key1[1].key3",
+		RawString: true,
+	}
+
+	body := `{"key1":[{"key2":"val2"},{"key3":"val3"}]}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, "val3", val)
+}
+
+func TestRenderer_Path_Json_Full(t *testing.T) {
+	t.Parallel()
+
+	renderer := column.NewRenderer()
+	c := column.Path{}
+
+	body := `{"key1":[{"key2":"val2","key3":"val3"}]}`
+	header := http.Header{"Content-Type": []string{"application/json"}}
+
+	val, err := renderer.CSVValue(c, recordctx.FromHTTP(time.Now(), &http.Request{Header: header, Body: io.NopCloser(strings.NewReader(body))}))
+	assert.NoError(t, err)
+	assert.Equal(t, `{"key1":[{"key2":"val2","key3":"val3"}]}`, val)
+}
+
+func TestRenderer_Path_Json_UndefinedKey_Error(t *testing.T) {
```

Comment:
I miss case with out of index. e.g `key1[2].key3` case that end's up with an error.